src/args.rs

```diff
@@ -150,3 +152,47 @@ pub fn parse_args() -> crate::MiniserveConfig {
         file_upload: args.file_upload,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn create_required_auth (username: &str, password: &str, encrypt: &str) -> auth::RequiredAuth {
+        use auth::*;
+        use RequiredAuthPassword::*;
+
+        RequiredAuth {
+            username: username.to_owned(),
+            password: match encrypt {
+                "plain" => Plain(password.to_owned()),
+                "sha256" => Sha256(password.to_owned()),
+                "sha512" => Sha512(password.to_owned()),
+                _ => panic!("Unknown encryption type")
+            },
+        }
+    }
+
+    #[test]
+    fn parse_auth_plain() {
+        assert_eq!(
+            parse_auth("username:password").unwrap(),
```

Comment:
You can use `?` in `#[test]` functions now.

Reply:
That would make the function returns an `Err` instead of panicking.

Reply:
Would this also mean that I have to specify return type?

![error](https://user-images.githubusercontent.com/11488886/56431788-45dafd80-62f5-11e9-9607-ea7cf2c3df6e.png)

Reply:
This would certainly make the code more verbose: 

![new code](https://user-images.githubusercontent.com/11488886/56432312-14fbc800-62f7-11e9-894c-eaadfb86f873.png)

Do you still insist on this change?

Reply:
Well, it depends. If you look at the integration, tests, you can see that we use quite a few `?`s where we would otherwise have multiple `.unwrap`s which is not quite as tidy in exchange for a single complicated `Result` per function. I think it's a fair trade-off and would rather roll with `?` for the time being.

Reply:
As you wish